tests/conftest.py

```diff
@@ -1,98 +1,194 @@
+
+def td_platform_fixture(td):
+    def platform_with_td(request):
+        type = request.param
+        postgres_dsn = request.config.option.postgres_dsn
+        bctx = get_backend_context(type, postgres_dsn)
+
+        with bctx as backend:
+            platform = Platform(_backend=backend)
+            td.load_dataset(platform)
+            yield platform
+
+    return platform_with_td
+
+
+# class scope fixture with big test data
+db_platform_big = pytest.fixture(
+    td_platform_fixture(big), scope="class", name="db_platform_big"
 )
 
+platform_med = pytest.fixture(
+    td_platform_fixture(medium), scope="class", name="platform_med"
+)
 
-@pytest.fixture(scope="function")
-def test_data_datetime():
-    return TEST_DF_DATETIME.copy()
-
-
-TEST_DF_CATEGORICAL = pd.DataFrame(
-    [
-        ["World", "Primary Energy", "EJ/yr", 2010, "A", 6.0],
-        ["World", "Primary Energy", "EJ/yr", 2010, "B", 3],
-        ["World", "Primary Energy", "EJ/yr", 2005, "A", 1],
-        ["World", "Primary Energy", "EJ/yr", 2005, "B", 0.5],
-    ],
-    columns=["region", "variable", "unit", "step_year", "step_category", "value"],
+rest_platform_med = pytest.fixture(
+    td_platform_fixture(medium), scope="class", name="rest_platform_med"
 )
 
 
-@pytest.fixture(scope="function")
-def test_data_categorical():
-    df = TEST_DF_CATEGORICAL.copy()
-    return df
-
-
-TEST_DF_ANNUAL = pd.DataFrame(
-    [
-        ["World", "Primary Energy", "EJ/yr", 2005, 1],
-        ["World", "Primary Energy", "EJ/yr", 2010, 6.0],
-        ["World", "Primary Energy|Coal", "EJ/yr", 2005, 0.5],
-        ["World", "Primary Energy|Coal", "EJ/yr", 2010, 3],
-    ],
-    columns=["region", "variable", "unit", "step_year", "value"],
-)
+def pytest_generate_tests(metafunc):
+    # This is called for every test. Only get/set command line arguments
+    # if the argument is specified in the list of test "fixturenames".
 
+    # parse '--backend' option
+    be_args = metafunc.config.option.backend.split(",")
+    backend_types = [t.strip() for t in be_args]
+    for bt in backend_types:
+        if bt not in backend_choices:
+            raise ProgrammingError(f"'{bt}' not a valid backend")
 
-@pytest.fixture(scope="function")
-def test_data_annual():
-    df = TEST_DF_ANNUAL.copy()
-    return df
+    # the `backend_fixtures` dict tells us which backends are allowed
+    # for which fixtures
+    for fixturename, allowed_types in backend_fixtures.items():
+        pres_types = [t for t in backend_types if t in allowed_types]
+        if fixturename in metafunc.fixturenames:
+            metafunc.parametrize(fixturename, pres_types, indirect=True)
 
 
 @pytest.fixture(scope="function")
 def profiled(request):
+    """Use this fixture for profiling tests:
+    ```
+    def test(profiled):
+        # setup() ...
+        with profiled():
+            complex_procedure()
+        # teardown() ...
+    ```
+    Profiler output will be written to '.profiles/{testname}.prof'
+    """
```

Comment:
This looks very handy, but I'm somewhat surprised that pytest doesn't offer something like this itself (I checked and couldn't find anything quickly, at least).